core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -438,6 +454,49 @@ private Get createGet(Snapshot.Key key) throws CrudException {
     return (Get) prepareStorageSelection(buildableGet.build());
   }
 
+  public void waitForRecoveryCompletionIfNecessary() throws CrudException {
+    for (RecoveryExecutor.Result recoveryResult : recoveryResults) {
+      try {
+        if (snapshot.containsKeyInWriteSet(recoveryResult.key)
+            || snapshot.containsKeyInDeleteSet(recoveryResult.key)
+            || snapshot.isValidationRequired()) {
+          recoveryResult.recoveryFuture.get();
```

Comment:
If the recovered records are in writeSet or deleteSet, or if serializable validation is required, wait for the lazy recoveries to complete.

Reply:
I think for records in writeSet or deleteSet, it's required for correctness. And in case where serializable validation is required, it's to avoid unnecessary aborts.

For records in the writeSet or deleteSet, if we don’t wait for lazy recovery to complete, we might end up preparing records whose status is still PREPARED or DELETED, which is not the intended behavior.

Similarly, when serializable validation is required, if we don’t wait for lazy recovery, the validation could fail if it reads records with PREPARED or DELETED status. That’s why waiting is necessary in these cases.

Reply:
> I think for records in writeSet or deleteSet, it's required for correctness.

@brfrn169 Can you give me a simple example for me?

Reply:
Based on the offline discussion, I understand that it is not for correctness, but a nice-to-have for unnecessary aborts.
Thanks!

Reply:
@brfrn169 Can you add a short comment about the background reason?